### DISCUSSION TITLE
Showing a window that is hidden using the keyboard?

### DISCUSSION THREAD

**Discussion by forte97** [bakkeby/dwm-flexipatch#269] [Q&A] [answered]
I was wondering if there was a way to show a hidden window using the keyboard, or selecting a hidden window maybe. Mod+Ctrl+Z is fine when only one window is open, but when multiple windows are open, I couldn't find a way to show a hidden window without clicking on the top bar.

**Comment by bakkeby** [comment, marked as answer]
There is an undocumented feature where you can use the normal focusstack to also focus on hidden windows by changing the argument to +/-2. 

```c
	{ MODKEY,                       XK_j,          focusstack,             {.i = +2 } },
	{ MODKEY,                       XK_k,          focusstack,             {.i = -2 } },
```

This allows you to select hidden windows in addition to normal visible windows. You can add separate keybindings for the above if you want.

It is worth noting that due to how hidden windows are in iconic state they won't be able to receive input focus, in this case the input focus will remain with the previously focused visible window.

The solution is not particularly good imho; it would be better to have a separate function that only focuses on hidden windows.

I'm not sure I remember exactly but I think that you should have visual feedback in the bar(s) when a hidden window has focus.

**Reply by forte97** [reply]
> There is an undocumented feature where you can use the normal focusstack to also focus on hidden windows by changing the argument to +/-2.
> 
> ```c
> 	{ MODKEY,                       XK_j,          focusstack,             {.i = +2 } },
> 	{ MODKEY,                       XK_k,          focusstack,             {.i = -2 } },
> ```
> 
> This allows you to select hidden windows in addition to normal visible windows. You can add separate keybindings for the above if you want.
> 
> It is worth noting that due to how hidden windows are in iconic state they won't be able to receive input focus, in this case the input focus will remain with the previously focused visible window.
> 
> The solution is not particularly good imho; it would be better to have a separate function that only focuses on hidden windows.
> 
> I'm not sure I remember exactly but I think that you should have visual feedback in the bar(s) when a hidden window has focus.

thanks, worked great
